### PR TITLE
util, test: Don't allow Base32/64-decoding or ParseMoney(…) on strings with embedded NUL characters. Add tests.  Add negative test case.

### PR DESCRIPTION
**Pull request by barton2526**
> Don't allow Base32/64-decoding or `ParseMoney(…)` on strings with embedded `NUL` characters.

Also test these cases

Ref: https://github.com/bitcoin/bitcoin/pull/17753

Add test case to ensure negative amounts fail `ParseMoney`

Ref: https://github.com/practicalswift/bitcoin/commit/fa3a81af18347a1d3fed41aa89ee643cbf0e7abc

Also adds copyright headers

### CODE REVIEW

File: src/test/base32_tests.cpp [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
+using namespace std::literals;
 
 BOOST_AUTO_TEST_SUITE(base32_tests)
 
 BOOST_AUTO_TEST_CASE(base32_testvectors)
 {
     static const std::string vstrIn[]  = {"","f","fo","foo","foob","fooba","foobar"};
     static const std::string vstrOut[] = {"","my======","mzxq====","mzxw6===","mzxw6yq=","mzxw6ytb","mzxw6ytboi======"};
-    for (unsigned int i = 0; i < std::size(vstrIn); i++)
+    for (unsigned int i=0; i<std::size(vstrIn); i++)
```
Comment (jamescowens) [inline src/test/base32_tests.cpp]: The original formatting was correct. Not the above.

Comment (barton2526) [reply src/test/base32_tests.cpp]: https://github.com/barton2526/Gridcoin-Research/commit/f9760d52b11de6f1924e1cb4403bba07b30a7484

**Review by jamescowens — APPROVED** [review]
